### DISCUSSION TITLE
input reference point is NOT included in the connectComponent

### DISCUSSION THREAD

**Discussion by MaryemBl** [insarlab/MintPy#819] [Q&A] [answered]
Hello everyone 
I'm working with SNAP to produce Mintpy inputs , I use snaphu to generate connected components, it works fine for some ifgram triplets, but not for the others, the error displayed is: 

_command line_: snaphu -G conncomp.img UnwPhase_ifg_VV_27Oct2020_20Nov2020.snaphu.img 1793(tried 495 also)
_error_ : snaphu v1.4.2
extra data in file UnwPhase_ifg_VV_27Oct2020_20Nov2020.snaphu.img (bad linelength?)

The header file:
ENVI
description = {Unwrapped Phase from complex data - Unit: 1}
samples = 1793
lines = 495
bands = 1
header offset = 0
file type = ENVI Standard
data type = 4
interleave = bsq
byte order = 0
band names = { UnwPhase_ifg_VV_27Oct2020_20Nov2020 }
data gain values = {1.0}
data offset values = {0.0}

Can enyone tell why this happened with this ifgram ?

I tried to apply mintpy for triplets with connected components (snaphu -G command worked on them), another error occured, see log file below. my question is : how to choose a reference point when using connected components?

[log.txt.txt](https://github.com/insarlab/MintPy/files/9154960/log.txt.txt)
Thank you

**Comment by hardreddata** [comment, marked as answer]
To your second question and `ValueError: input reference point is in masked OUT area defined by maskConnComp.h5!` you could disable the masking.
```
mintpy.reference.maskFile      = no
```
The alternative is to set the reference point to be a location inside the connected component mask. 

Cheers.

**Reply by Karthi-Wolf** [reply]
Hey hiii, I also got the similar error. i got 'input reference point is in masked OUT area defined by maskConnComp.h5!' . can you explain why it happens and how setting mintpy.reference.maskFile      = no helps us? Thanks in advance....

**Reply by david-ncu2019** [reply]
In my experience, only the process of phase unwrapping error correction needs the reference point to be located in `maskConnComp.h5`.

**Reply by Karthi-Wolf** [reply]
Okay mr. david, thanks for letting me know.

**Comment by david-ncu2019** [comment]
Hello everyone,

I have the same issue, which prevents me from correcting my phase unwrapping errors.

When I ran `generate_mask.py inputs/ifgramStack.h5 --nonzero -o maskConnComp.h5`, the program produced a mask entirely filled with 0.

I don't know why the algorithm did not correct only the selected interferograms and instead ignored the ones that had been excluded after the step `modify_network`.

**Reply by Karthi-Wolf** [reply]
Did you solved this issue, if it so please let me know, i'm also in that stage where my whole image of maskConnComp.h5 is of 0 pixel values.

**Reply by david-ncu2019** [reply]
> Did you solved this issue, if it so please let me know, i'm also in that stage where my whole image of maskConnComp.h5 is of 0 pixel values.

Step 1: you may view the maskConnComp.h5 using the command `view.py maskConnComp.h5 --zm` (using zero-mask mode). If the mask is all 0, move to the next step.

Step 2: You can use `view.py inputs/ifgramStack.h5 *coherence* -c turbo` to check if there are any problematic interferograms. Let the function plot all the figures, even if there are thousands of them, and check the interferograms with your own eyes. The entire zero-value maskConnComp.h5 file looks a little strange.

Why check coherence? Because the coherence map is clear and makes it easier for us to detect any problems.

Please let me know if you can solve the issue.

**Reply by Karthi-Wolf** [reply]
Thank you so much for this reply, in my coherence images (424)i making a mistake if letting some interferograms with burst registration issues. i am getting a straight line going horizontally. here i attach the images, please kindly guide me.

<img width="1152" height="635" alt="velocity" src="https://github.com/user-attachments/assets/10aeb765-8e4f-430e-bfe5-3c7df1b2dd8b" />
<img width="714" height="619" alt="temporal_coherence" src="https://github.com/user-attachments/assets/f06c0844-0924-4b0f-b2c2-bfa81e8ff069" />
<img width="1148" height="537" alt="geometry_radar" src="https://github.com/user-attachments/assets/cd05d313-700b-4a65-bc33-853e5d46ce99" />
<img width="808" height="350" alt="maskConnComp" src="https://github.com/user-attachments/assets/315e8cfd-9306-403b-8a1e-cd24c8d02b12" />
<img width="2128" height="856" alt="coherence_9" src="https://github.com/user-attachments/assets/e016e14c-d893-4f83-b7aa-47528f7e41ce" />
<img width="2123" height="1184" alt="coherence_1" src="https://github.com/user-attachments/assets/7b4ee7b5-666b-431e-a8fc-f363531adce9" />
<img width="1113" height="553" alt="geo_velocity" src="https://github.com/user-attachments/assets/7a5016d5-cf9a-4a85-9adf-f8daf2fb11c7" />

**Reply by david-ncu2019** [reply]
It is really clear that there is something wrong with the interferograms. A clear white line crosses your study area horizontally (inteferograms), and another line cuts vertically through the image (velocity).

**Reply by Karthi-Wolf** [reply]
Hi mr. david, you are right, after some analysis I gt to know the reason behind the vertical line, it's becasue it's the area that spans over the given DEM file. I gave my bounding box t be inside the DEM, but as we know ISCE takes little extra area to process so the line cross exactly at the longitude 86.00 as my DEM is 84-85 lon  (upto 86). you can see it in the 3rd image 2.height. It's my DEM. 
So, now I have seen my unwrap phase and unwrap phase wrap , the horizontal line alomost present in all the images. What should I do to get the output?
I very much appreciate your help in this error. 
<img width="2166" height="1184" alt="unwrapPhase_2" src="https://github.com/user-attachments/assets/9a28d3c8-72ef-43c5-bec2-c2b6f91851ff" />
<img width="2138" height="1184" alt="unwrapPhase_wrap_2" src="https://github.com/user-attachments/assets/1f2192d5-5432-41d1-9351-310f8f986a01" />

**Reply by ditafaith** [reply]
Hi Karthi,
Thanks for sharing this issue. I worried about why ISCE processes beyond the area defined by the bounding box?
This condition makes us provide a larger DEM (says 1 degree wider).

Also, you can crop the AOI using MintPy.

**Reply by Karthi-Wolf** [reply]
Yes, we can do like that thanks for the suggestion and my study area is safer from that. How can i avoid the horizontal line (it's passes through my AOI) i have already dropped around 700+ pairs with the network.minCoherence in network modification step and I am clueless to find the error source!. thanks in advance.

**Reply by david-ncu2019** [reply]
That horizontal line may unexpectedly come from the process of burst merging. Where is your study area? What bursts did you use to process?

**Comment by Karthi-Wolf** [comment]
hi mr.david, this is my study area , a eastern coastal town in India. <img width="849" height="558" alt="image" src="https://github.com/user-attachments/assets/7552a004-9237-4456-8388-7c743047a6b6" />
I have downloaded the whole sentinel image and processed with 'stackSentinel.py -s .....slcs -o .....orbits -d .....dem/dem.wgs84 -b '19.760 19.973 85.718 85.976'  -W interferogram -n '1 2 3' -a ......./orbits -c 5 -C NESD -e 0.70 -m 20160101'

**Comment by sharkbig** [comment]
I've also encountered the same problem, but the way that set the mask file to "no" doesn't work for me. 
Thus, I try to assign the reference point manually according to the maskConnComp.h5 file, and successful proceed to the next step.
